doc/mainstay.md

```diff
@@ -0,0 +1,127 @@
+# Mainstay integration
+
+Options for attesting state to a mainstay proof-of-publication service. 
+
+Assumptions: 
+
+Mainstay service is available over http interface (or via SOCKS5 Tor proxy). 
+Mainstay service is available and funded with a valid `token_id` for verifiation. 
+Funding (via LN payment) is performed in advance and out of band for subscrption. (i.e. `token_id` is already performed.)
```

Comment:
This is a fundamental issue with mainstay (or any proof-of-publication mechanism or service). The commitments must be provably unique in a given publication space, and so user A must have exclusive access to their own publication space (i.e. 'slot' in mainstay), necessitating user credentials. The credentials can be updated, but the identification of the publication space they are linked to cannot be - the service will always know it's the same user posting commitments to the same slot. 

But I don't think there is an issue with this privacy-wise. The user can blind the commitments themselves if required, and store the blinding nonces with the proofs for verification.

Reply:
Okay I see the provably unique requirement in a proof-of-publication, though on the exclusive access I wonder if a user signature (and therefore posession of a secret key) could be included in the commitment scope. If you have duplication or equivocation of a publication space it can be disregarded at both client / server level. If my understanding of proof-of-publication space is correct.

Otherwise yes credential can be re-used indefinitely by the user, like the service provider binds a slot at the first credential redemption, and allow re-use of it.